api/tests/authentication_tests.py

```diff
@@ -0,0 +1,89 @@
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(self.auth, get_addresses_req) == None
+
+    def test_no_token_returns_none(self):
+        headers = {'HTTP_AUTHORIZATION': 'Bearer '}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(self.auth, get_addresses_req) == None
+
+    @responses.activate()
+    def test_non_200_resp_from_fxa_returns_none(self):
+        responses.add(
+            responses.POST, self.fxa_verify_path, status=401, json={'error':'401'}
+        )
+        not_found_token = 'not-found-123'
+        headers = {'HTTP_AUTHORIZATION': f'Bearer {not_found_token}'}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(self.auth, get_addresses_req) == None
+        print(f'asserting {get_cache_key(not_found_token)} in {cache}')
+        assert cache.get(get_cache_key(not_found_token)) is not None
+
+    @responses.activate()
+    def test_200_resp_from_fxa_inactive_token_returns_none(self):
+        responses.add(
+            responses.POST, self.fxa_verify_path, status=200,
+            json={'active': False}
+        )
+        inactive_token = 'inactive-123'
+        headers = {'HTTP_AUTHORIZATION': f'Bearer {inactive_token}'}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(self.auth, get_addresses_req) == None
+        assert cache.get(get_cache_key(inactive_token)) is not None
+
+    @responses.activate()
+    def test_200_resp_from_fxa_no_matching_user_returns_none(self):
+        responses.add(
+            responses.POST, self.fxa_verify_path, status=200,
+            json={'active': True, 'sub': 'not-a-relay-user'}
+        )
+        non_user_token = 'non-user-123'
+        headers = {'HTTP_AUTHORIZATION': f'Bearer {non_user_token}'}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(self.auth, get_addresses_req) == None
+        assert cache.get(get_cache_key(non_user_token)) is not None
+
+    @responses.activate()
+    def test_200_resp_from_fxa_for_user_returns_user(self):
+        self.sa = baker.make(SocialAccount, uid=self.uid, provider='fxa')
+        user_token = 'user-123'
+        now_time = int(datetime.now().timestamp())
+        exp_time = now_time + 60*60
+        responses.add(
+            responses.POST, self.fxa_verify_path, status=200,
+            json={'active': True, 'sub': self.uid, 'exp': exp_time}
+        )
+
+        headers = {'HTTP_AUTHORIZATION': f'Bearer {user_token}'}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        assert self.auth.authenticate(
+            self.auth, get_addresses_req
+        ) == (self.sa.user, None)
+        print(f'asserting {get_cache_key(user_token)} in {cache}')
+        assert cache.get(get_cache_key(user_token)) is not None
```

Comment:
This is causing [the test failure](https://app.circleci.com/pipelines/github/mozilla/fx-private-relay/3929/workflows/a7247be0-9e12-4567-ab14-cb5d71800bc5/jobs/7623) that I don't understand. 😢 
```
>       assert cache.get(get_cache_key(user_token)) is not None
E       AssertionError: assert None is not None
E        +  where None = <bound method LocMemCache.get of <django.core.cache.backends.locmem.LocMemCache object at 0x7f0a05e3be50>>('fxa_token_user-123')
E        +    where <bound method LocMemCache.get of <django.core.cache.backends.locmem.LocMemCache object at 0x7f0a05e3be50>> = <django.core.cache.DefaultCacheProxy object at 0x7f0a088b26d0>.get
E        +    and   'fxa_token_user-123' = get_cache_key('user-123')
```